### PR TITLE
Add client diagnostics api

### PR DESCRIPTION
**Pull request by GalRogozinski**
# Description of change

Add client api functionality to internal diagnostics api.
Fix missing fields in message diagnostic csv.
Part of #1253 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Added integration tests

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: plugins/webapi/tools/plugin.go [hunk 1]
```diff
@@ -13,6 +13,21 @@ import (
...
+	RouteDiagnosticsFirstWeakMessageReferences = RouteDiagnosticMessages + "/firstweakreferences"
+	RouteDiagnosticsMessageRank                = RouteDiagnosticMessages + "/rank/:rank"
+	RouteDiagnosticsUtxoDag                    = routeDiagnostics + "/utxodag"
+	RouteDiagnosticsBranches                   = routeDiagnostics + "/branches"
+	RouteDiagnosticsLazyBookedBranches         = RouteDiagnosticsBranches + "/lazybooked"
+	RouteDiagnosticsInvalidBranches            = RouteDiagnosticsBranches + "/invalid"
+	RouteDiagnosticsTips                       = routeDiagnostics + "/tips"
+	RouteDiagnosticsStrongTips                 = RouteDiagnosticsTips + "/strong"
+	RouteDiagnosticsWeakTips                   = RouteDiagnosticsTips + "/weak"
+	RouteDiagnosticsDRNG                       = routeDiagnostics + "/drng"
```
Comment (capossele) [inline plugins/webapi/tools/plugin.go:28]: Please add a comment to each of those to make the go linter happy or, in alternative, make them unexported

**Review by capossele — CHANGES_REQUESTED** [review]

File: plugins/webapi/tools/plugin.go [hunk 2]
```diff
@@ -13,6 +13,32 @@ import (
 // PluginName is the name of the web API tools endpoint plugin.
 const PluginName = "WebAPI tools Endpoint"
 
+const (
+	routeDiagnostics = "tools/diagnostic"
+	// API route for message diagnostics
```
Comment (capossele) [inline plugins/webapi/tools/plugin.go:18]:
```suggestion
	// RouteDiagnosticMessages is the API route for message diagnostics
```

Comment (capossele) [reply plugins/webapi/tools/plugin.go:18]: the following should also start with the name of the exported constant as the dog suggest 🐶

**Review by capossele — CHANGES_REQUESTED** [review]

File: plugins/webapi/tools/plugin.go (hunk 2 above)
Comment (GalRogozinski) [reply plugins/webapi/tools/plugin.go]: dog has been fed

**Review by capossele — APPROVED** [review]

**Pull request merged**